File: Comp-u-Lock-Web/app/models/account.rb
class Account < ActiveRecord::Base
  attr_accessible :computer_id, :allotted_time, :domain, 
  :tracking, :used_time, :username, :admin,  :account_process_attributes, :restriction_attributes

  validates :username, :presence => true
  validates :computer_id, :presence => true
  
  belongs_to :computer

  
  has_many :account_process, :class_name => "AccountProcess", :dependent => :destroy
  

  has_many :restriction, dependent: :destroy

  accepts_nested_attributes_for :account_process, :restriction

  def as_json options={}
    {
      id: id,
      computer_id: computer_id,
      domain: domain,
      username: username,
      tracking: tracking,
      admin: admin,
      allotted_time: allotted_time,
      used_time: used_time,
      
      account_process_attributes: account_process,

      restriction_attributes: restriction,

      created_at: created_at,
      update_at: updated_at
    }
  end
end

File: Comp-u-Lock-Desktop/Database/Models/Account.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Database.Models
{
    [JsonObject]
    public class Account
    {
        public Account()
        {
        }

        [JsonIgnore]
        public int Id { get; set; }
        [JsonProperty(PropertyName = "id", DefaultValueHandling = DefaultValueHandling.Ignore)]
        public int WebId { get; set; }

        [JsonProperty(PropertyName = "computer_id")]
        public int ComputerId { get; set; }
        [JsonProperty(PropertyName = "domain")]
        public string Domain { get; set; }
        [JsonProperty(PropertyName = "username")]
        public string Username { get; set; }

        [JsonProperty(PropertyName = "admin", NullValueHandling = NullValueHandling.Ignore)]
        public bool Admin { get; set; }

        [JsonIgnore]
        public bool Locked { get; set; }

        [JsonProperty(PropertyName = "tracking")]
        public bool Tracking { get; set; }
        [JsonProperty(PropertyName = "allotted_time")]
        public TimeSpan AllottedTime { get; set; }
        [JsonProperty(PropertyName = "used_time")]
        public TimeSpan UsedTime { get; set; }

        [JsonProperty(PropertyName = "account_process_attributes", NullValueHandling = NullValueHandling.Ignore)]
        public List<Process> Processes { get; set; }

        [JsonProperty(PropertyName = "restriction_attributes", NullValueHandling = NullValueHandling.Ignore)]
        public List<Restriction> Restrictions { get; set; }

        [JsonProperty(PropertyName = "created_at")]
        public DateTime CreatedAt { get; set; }
        [JsonProperty(PropertyName = "updated_at")]
        public DateTime UpdatedAt { get; set; }

        public string ToJSON()
        {
            return JsonConvert.SerializeObject(this);
        }
    }
}
